core/tx_pool.go

```diff
@@ -593,6 +667,15 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 		return ErrInvalidSender
 	}
 
+	if sorted_oracles.GetAddress() != nil && tx.To() != nil && tx.Data() != nil && sorted_oracles.GetAddress().Hex() == tx.To().Hex() {
```

Comment:
Oracles and txpool are already coupled because txpool compares gas prices with different currencies. The check for which transactions are prioritized could be more general, but that's probably out of scope for this PR.

Made a new function to check if tx is a priority tx.